rocksdb_replicator/replicated_db.cpp

```diff
@@ -317,7 +331,23 @@ void RocksDBReplicator::ReplicatedDB::pullFromUpstream() {
           }
 
           if (!response.updates.empty()) {
+            db->pullFromUpstreamNoUpdates_ = 0;
             db->cond_var_.notifyAll();
+          } else {
+            incCounter(kReplicatorPullRequestsNoUpdates, 1, db->db_name_);
+            // no updates consecutively, and the upstream says it's NOT a leader.
+            // Therefore we reset upstream.
+            db->pullFromUpstreamNoUpdates_++;
+            if (response.role != ReplicaRole::LEADER
```

Comment:
Do you want to add a counter to record pull from Non-leader for replication?

Reply:
good idea, added `kReplicatorPullFromNonLeader`